OgreMain/src/OgreSceneNode.cpp

```diff
@@ -194,6 +194,15 @@ namespace Ogre {
         needUpdate();
     }
     //-----------------------------------------------------------------------
+    void SceneNode::destroyAllObjects(void)
+    {
+        while (getAttachedObjects().size() > 0) {
```

Comment:
please use `empty()`and `front()` as below: https://github.com/OGRECave/ogre/pull/2963/files#diff-8b29158c3b96916d83c3b4bbc1c9ee083fee97bff80ae347e6401c6bc49dd5afR358